src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/QualityGuidelines/AvoidMultipleEnumerationsTests.cs

```diff
@@ -0,0 +1,3144 @@
+using System.Collections.Generic;
+using System.Linq;
+
+public class Bar
+{
+    public void Sub(int[] i)
+    {
+        var j = Enumerable.Range(1, 10);
+        var z = [|j|].AsEnumerable();
+        z.AsEnumerable().First();
+        z.First();
+
+        var x = i.AsEnumerable();
+        i.AsEnumerable().First();
+        i.First();
+    }
+}";
+            await VerifyCS.VerifyAnalyzerAsync(csharpCode);
+
+            var vbCode = @"
+Imports System.Collections.Generic
+Imports System.Linq
+
+Namespace NS
+    Public Class Bar
+        Public Sub Goo(i As Integer())
+            Dim j = Enumerable.Range(1, 10)
+            Dim z = [|j|].AsEnumerable()
+            z.AsEnumerable().ElementAt(10)
+            z.ElementAt(10)
+            
+            Dim x = i.AsEnumerable()
+            x.AsEnumerable().ElementAt(10)
+            x.ElementAt(10)
+        End Sub
+    End Class
+End Namespace";
+
+            await VerifyVB.VerifyAnalyzerAsync(vbCode);
+        }
+
+        [Fact]
+        public async Task TestAsEnumerable2()
+        {
+            var csharpCode = @"
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+public class Bar
+{
+    public void Sub()
+    {
+        var j = Enumerable.Range(1, 10);
+        var z = [|j|].AsEnumerable();
+        z.First();
+        z.First();
+    }
+}";
+            await VerifyCS.VerifyAnalyzerAsync(csharpCode);
+
+            var vbCode = @"
+Imports System.Collections.Generic
+Imports System.Linq
+
+Namespace NS
+    Public Class Bar
+        Public Sub Goo()
+            Dim j = Enumerable.Range(1, 10)
+            Dim z = [|j|].AsEnumerable()
+            z.First()
+            z.First()
+        End Sub
+    End Class
+End Namespace";
+
+            await VerifyVB.VerifyAnalyzerAsync(vbCode);
+        }
+
+        [Fact]
+        public async Task TestZip()
+        {
+            var csharpCode1 = @"
+using System.Collections.Generic;
+using System.Linq;
+
+public class Bar
+{
+    public void Sub(IEnumerable<int> j)
+    {
+        IEnumerable<int> i = Enumerable.Range(1, 10);
+        var k = [|i|].Zip(j, (a, b) => (a, b));
+        k.First();
+        [|i|].First();
+    }
+}";
+            await VerifyCS.VerifyAnalyzerAsync(csharpCode1);
+
+            var vbCode = @"
+Imports System.Collections.Generic
+Imports System.Linq
+
+Namespace Ns
+    Public Class Hoo
+        Public Sub Goo(j As IEnumerable(Of Integer))
+            Dim i = Enumerable.Range(1, 10)
+            Dim k = [|i|].Zip(j, Function(a, b) (a, b))
+            k.First()
+            [|i|].First()
+        End Sub
+    End Class
+End Namespace
+";
+            await VerifyVB.VerifyAnalyzerAsync(vbCode);
+        }
+
+        [Fact]
+        public void TestNet6AddedMethod()
+        {
+            // newly added method in .Net 6, currently can't found by unit tests
+            Assert.Contains("MaxBy", AvoidMultipleEnumerations.s_enumeratedParametersLinqMethods);
```

Comment:
You can test APIs in .Net 6 https://github.com/dotnet/roslyn-analyzers/blob/107d82a590a1c832f048d4f4bf1df61fd2fb3ec8/src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/DetectPreviewFeatureUnitTests.Misc.cs#L21